src/Ast/Expression.elm

```diff
@@ -115,97 +116,119 @@ string =
     in
-        multiString <|> singleString
+        withMeta <| multiString <|> singleString
 
 
 integer : Parser s Expression
 integer =
-    Integer <$> Combine.Num.int
+    withMeta <| Integer <$> Combine.Num.int
 
 
 float : Parser s Expression
 float =
-    Float <$> Combine.Num.float
+    withMeta <| Float <$> Combine.Num.float
 
 
 access : Parser s Expression
 access =
-    Access <$> variable <*> many1 (Combine.string "." *> loName)
+    withMeta <| Access <$> variable <*> many1 (Combine.string "." *> loName)
 
 
 accessFunction : Parser s Expression
 accessFunction =
-    AccessFunction <$> (Combine.string "." *> loName)
+    withMeta <| AccessFunction <$> (Combine.string "." *> loName)
 
 
 variable : Parser s Expression
 variable =
-    Variable
-        <$> choice
-                [ singleton <$> loName
-                , sepBy1 (Combine.string ".") upName
-                , singleton <$> parens operator
-                , singleton <$> parens (Combine.regex ",+")
-                , singleton <$> emptyTuple
-                ]
+    withMeta <|
+        Variable
+            <$> choice
+                    [ singleton <$> loName
+                    , sepBy1 (Combine.string ".") upName
+                    , singleton <$> parens operatorAsString
+                    , singleton <$> parens (Combine.regex ",+")
+                    , singleton <$> emptyTuple
+                    ]
 
 
 list : OpTable -> Parser s Expression
 list ops =
     lazy <|
         \() ->
-            List <$> brackets (commaSeparated_ <| expression ops)
+            withMeta <| List <$> brackets (commaSeparated_ <| expression ops)
 
 
 tuple : OpTable -> Parser s Expression
 tuple ops =
     lazy <|
         \() ->
-            Tuple
-                <$> (parens (commaSeparated_ <| expression ops)
-                        >>= \a ->
-                                case a of
-                                    [ _ ] ->
-                                        fail "No single tuples"
-
-                                    anyOther ->
-                                        succeed anyOther
-                    )
+            withMeta <|
+                Tuple
+                    <$> (parens (commaSeparated_ <| expression ops)
+                            >>= \a ->
+                                    case a of
+                                        [ _ ] ->
+                                            fail "No single tuples"
+
+                                        anyOther ->
+                                            succeed anyOther
+                        )
 
 
 record : OpTable -> Parser s Expression
 record ops =
     lazy <|
         \() ->
-            Record <$> braces (commaSeparated ((,) <$> loName <*> (symbol "=" *> expression ops)))
+            withMeta <|
+                Record
+                    <$> braces
+                            (commaSeparated
+                                ((,)
```

Comment:
Please try to avoid splitting arguments to next line
Backpipe (<|) twice could do the job here